keyboards/sofle_pico/keymaps/default/keymap.c

```diff
@@ -0,0 +1,161 @@
+        * |      |      |      |      |      |      |-------|    |-------|      | PREV | PLAY | NEXT |      |      |
+        * `-----------------------------------------/       /     \      \-----------------------------------------'
+        *            | LGUI | LAlt | LCTR |LOWER | /Enter  /       \Space \  |RAISE | RCTR | RAlt | RGUI |
+        *            |      |      |      |      |/       /         \      \ |      |      |      |      |
+        *            `----------------------------------'           '------''---------------------------'
+        */
+        [_ADJUST] = LAYOUT(
+        XXXXXXX , XXXXXXX, XXXXXXX , XXXXXXX, XXXXXXX, XXXXXXX,                       XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+        QK_BOOT , XXXXXXX, XXXXXXX , XXXXXXX, CG_TOGG, XXXXXXX,                       XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+        XXXXXXX , XXXXXXX, CG_TOGG , XXXXXXX, XXXXXXX, XXXXXXX,                       XXXXXXX, KC_VOLD, KC_MUTE, KC_VOLU, XXXXXXX, XXXXXXX,
+        XXXXXXX , XXXXXXX, XXXXXXX , XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,     XXXXXXX, XXXXXXX, KC_MPRV, KC_MPLY, KC_MNXT, XXXXXXX, XXXXXXX,
+                            _______, _______, _______, _______, _______,     _______, _______, _______, _______, _______
+        )
+    // clang-format on
+};
+
+/* Default timeout for displaying logo on boot. */
+#ifndef OLED_LOGO_TIMEOUT
+/* 10 second timeout. */
+#    define OLED_LOGO_TIMEOUT 10000
+#endif
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+    return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case KC_COPY:
+            if (record->event.pressed) {
+                register_mods(mod_config(MOD_LCTL));
+                register_code(KC_C);
+            } else {
+                unregister_mods(mod_config(MOD_LCTL));
+                unregister_code(KC_C);
+            }
+            return false;
+        case KC_PASTE:
+            if (record->event.pressed) {
+                register_mods(mod_config(MOD_LCTL));
+                register_code(KC_V);
+            } else {
+                unregister_mods(mod_config(MOD_LCTL));
+                unregister_code(KC_V);
+            }
+            return false;
+        case KC_CUT:
+            if (record->event.pressed) {
+                register_mods(mod_config(MOD_LCTL));
+                register_code(KC_X);
+            } else {
+                unregister_mods(mod_config(MOD_LCTL));
+                unregister_code(KC_X);
+            }
+            return false;
+            break;
+        case KC_UNDO:
+            if (record->event.pressed) {
+                register_mods(mod_config(MOD_LCTL));
+                register_code(KC_Z);
+            } else {
+                unregister_mods(mod_config(MOD_LCTL));
+                unregister_code(KC_Z);
+            }
```

Comment:
None of these need a custom macro.  Copy can be set as `C(KC_C)` in the keymap, directly, for instance.

Reply:
Thank you! I've updated accordingly.